### ISSUE TITLE
optimize training

### ISSUE THREAD

**Issue by AleKitsune98** [urchade/GLiNER#91]
How to specify parameters like warmup_ratio, dropout in finetuning?

**Comment by mkmohangb** [comment]
Check this [file](https://github.com/urchade/GLiNER/blob/main/train.py) for the details.

**Comment by AleKitsune98** [comment]
thank you, I have tried the script but when I come to the val step it says "Object of type SimpleNameSpace is not JSONSerializable"

**Comment by urchade** [comment]
Hi @AleKitsune98, can you send me a screenshot of the error?

**Comment by mkmohangb** [comment]
I was able to reproduce the issue and created a [pull request](https://github.com/urchade/GLiNER/pull/92) with the fix.

**Comment by AleKitsune98** [comment]
Thank you very much, now I am getting division by zero during evaluation. I'm using training with huggingface model in config.yaml

**Comment by urchade** [comment]
Have you downloaded the correct dataset ? @AleKitsune98

**Comment by AleKitsune98** [comment]
I created a custom dataset following the format of sample_data.json

**Comment by urchade** [comment]
The training evaluation is intended for benchmark comparison on the data I linked before, but you can modify the code by looking at [this link](https://github.com/urchade/GLiNER/blob/main/examples/finetuning/trainer.py) if you have JSON evaluation data.

**Comment by AleKitsune98** [comment]
Thank you very much. I will take a look at it. I appreciate your work!
